tests/analysis_agent.py

```diff
@@ -0,0 +1,176 @@
+from cromwell_tools import api as cwm_api
+from cromwell_tools import cromwell_auth as cwm_auth
+from .analysis_entities import Workflow
+from contextlib import contextmanager
+import logging
+
+
+class AnalysisAgent:
+    def __init__(self, deployment, service_account_key):
+        """Agent model for talking to the HCA DCP Secondary-analysis service.
+
+        This model is designed to talk to the HCA DCP Secondary-analysis service, especially the underlying Cromwell
+            Workflow Execution Engine's API without exposing too many technical details to the users.
+
+        Args:
+            deployment (str): String representing the deployment environment to be queried on. It could be:
+                "dev", "integration", "staging" or "prod".
+            service_account_key (str): Optional, path to the service account JSON key file, which is required by
+                authenticate with the Secondary-analysis service (Crowmell API). If not provided, the agent will assume
+                no authentication required for talking to secondary analysis, which is very likely to break or skip
+                a lot of commands that are using this agent.
+                TODO: Add OAuth support to this agent so that users could authenticate through Google/Auth0.
+        """
+        self.deployment = deployment
+        self.cromwell_url = 'https://cromwell.caas-prod.broadinstitute.org'
+        self.cromwell_collection = 'lira-test' if self.deployment == 'integration' else f'lira-{self.deployment}'
+        self.auth = self._get_auth(service_account_key)
+
+    def _get_auth(self, service_account_key):
+        """Helper function to generate the auth object to talk to Secondary-analysis service."""
+        return cwm_auth.CromwellAuth.harmonize_credentials(service_account_key=service_account_key,
+                                                           url=self.cromwell_url)
+
+    @staticmethod
+    @contextmanager
+    def ignore_logging_msg(highest_level=logging.CRITICAL):
+        """A context manager that will prevent any logging messages triggered during the body from being processed.
+
+        Referred to https://gist.github.com/simon-weber/7853144
+        two kind-of hacks here:
+            * can't get the highest logging level in effect => delegate to the user
+            * can't get the current module-level override => use an undocumented (but non-private!) interface
+
+        highest_level (int): The maximum logging level in use. This would only need to be changed if
+            a custom level greater than CRITICAL is defined.
+        """
+        previous_level = logging.root.manager.disable
+
+        logging.disable(highest_level)
+
+        try:
+            yield
+        finally:
+            logging.disable(previous_level)
+
+    def query_by_workflow_uuid(self, uuid):
```

Comment:
Is it possible to re-use the dcp-diag tool here since the code is very similar?

Reply:
I copied the code from the dcp-diag and made slight changes. Unfortunately the only way to re-use the dcp-diag tool is to install it and a lot its own dependencies in this test suite, which seems burdensome. This is a consequence of that the dcp-diag tool was designed to be a CLI not library :( 

On this topic, I'm thinking maybe we could create a very light-weight "secondary-analysis" library and use it everywhere (e.g. dcp-diag, dcp, etc..). And cromwell-tools might be a good candidate for hosting it. What do you think?